src/datanode/src/datanode.rs

```diff
@@ -468,39 +467,46 @@ async fn open_all_regions(
         }
     }
     info!("going to open {} region(s)", regions.len());
-    let semaphore = Arc::new(tokio::sync::Semaphore::new(OPEN_REGION_PARALLELISM));
-    let mut tasks = vec![];
 
-    let region_server_ref = &region_server;
+    let mut region_requests = Vec::with_capacity(regions.len());
+    let mut region_ids = Vec::with_capacity(regions.len());
```

Comment:
It seems unnecessary, the `handle_batch_open_requests` method will return region ids.